community: Migrate Vertex AI Search Retriever from v1beta to v1

lgesuellip opened the pull request:
## PR Description

Hey team, this PR migrates the Vertex AI Search Retriever from v1beta to the stable v1 API.

## Type

🧹 Refactoring
✅ Test

## Changes

 Key changes:

- Updated imports from `discoveryengine_v1beta` to `discoveryengine_v1` 
- Removed unsupported v1beta features from `VertexAISearchRetriever`: embedding_spec, ranking_expression, relevance_threshold, personalization_spec, and natural_language_query_understanding_spec
- Removed custom embedding functionality dependent on beta features

## Testing

- Modified corresponding tests to exclude beta feature tests for `vertex_ai_search`.

Thank you Team,

lgesuellip commented on libs/community/langchain_google_community/vertex_ai_search.py:
```diff
@@ -279,23 +278,6 @@ class VertexAISearchRetriever(BaseRetriever, _BaseVertexAISearchRetriever):
     https://cloud.google.com/generative-ai-app-builder/docs/boost-search-results
     https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1beta/BoostSpec
     """
-    custom_embedding: Optional[Embeddings] = None
```
Hi @lkuligin , 

The main reason for this change is stability. When using v1beta, the retrieval times are inconsistent, sometimes taking up to 10x longer than v1. For example, there were cases where fetching results took 20 seconds instead of the expected 0.5 seconds. This kind of delay is not acceptable for a tool designed to support an agent efficiently.

Do you think we should maintain compatibility by supporting both the beta and stable versions in the class, or is it better to fully transition to the stable version?

lgesuellip replied:
@lkuligin let me know your thoughts, and we can align on the best approach moving forward.

lkuligin replied:
I'd suggest to maintain both clients, and if beta arguments are set, throw a warning and use v1beta client in that case (the default one should be v1).
Thanks!

lgesuellip commented:
Hey @lkuligin ,

I've added all the changes to support both the stable and beta versions of `vertex_ai_search` and updated the corresponding tests.

The changes turned out to be larger than expected.
Please let me know if you have any comments or suggestions.

Thanks!

lgesuellip commented:
@lkuligin ,

The integration tests are failing, but I’m unable to see the CI log details. Could you help me check?

Thanks so much!

lgesuellip commented:
@lkuligin

Please let me know if everything looks good or if there’s anything else needed. 
Thank you!

lkuligin commented on libs/community/pyproject.toml:
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "langchain-google-community"
-version = "2.0.3"
+version = "2.1.0"
```
don't change the version, please
releases go through a separate process

lkuligin commented on libs/community/langchain_google_community/vertex_ai_search.py:
```diff
@@ -25,13 +24,29 @@
-        SearchServiceClient,
+    from google.cloud.discoveryengine_v1 import (  # noqa: I001
+        ConversationalSearchServiceClient as StableConversationalSearchServiceClient,
+        SearchRequest as StableSearchRequest,
```
nits: let's call it `V1SearchRequest` and `V1BetaSearchRequest` for a better readability, please

or `SearchRequest` vs `BetaSearchRequest` maybe, but let's definitely drop `Stable`

lkuligin commented on libs/community/langchain_google_community/vertex_ai_search.py:
```diff
@@ -56,6 +71,8 @@ class _BaseVertexAISearchRetriever(Serializable):
     1 - Structured data
     2 - Website data
     """
+    beta: bool = False
```
can we make it a private attribute too, please?

lkuligin commented on libs/community/langchain_google_community/vertex_ai_search.py:
```diff
@@ -337,10 +386,71 @@ def __init__(self, **kwargs: Any) -> None:
+                    "Please provide a custom_embedding_ratio if you provide a "
+                    "custom embedding model or a custom_embedding_field_path."
+                )
+            if not 0 <= self.custom_embedding_ratio <= 1:
```
this is not needed, pydantic will handle it because fo your change

lgesuellip replied:
You are right!

lgesuellip commented:
@lkuligin Thank you for the feedback! All the changes pushed.

lkuligin commented:
@lgesuellip thanks for the changes! some linting issues remain, take a look, please (probably it's enough just to ignore the warnings for now).

lgesuellip commented:
> * Removed unsupported v1beta features from `VertexAISearchRetriever`: embedding_spec, ranking_expression, relevance_threshold, personalization_spec, and natural_language_query_understanding_spec

@lkuligin I just updated it! Thanks,
